src/transforms/api/_transform.py

```diff
@@ -186,8 +240,9 @@ class TransformInput:
     """TransformInput class, passed when using @transform decorator."""
 
     def __init__(self, input_arg: Input):
-        self._dataframe = input_arg.dataframe()
+        self._input_arg = input_arg
         self._dataset_identity = input_arg.get_dataset_identity()
+        self._local_path = input_arg.get_local_path_to_dataset()
```

Comment:
I thought about this again and right now it will trigger a download already in the constructor here, no-matter if it's used or not. 
Maybe you can directly call `self._input_arg.get_local_path_to_dataset()` in the filesystem() function and not store `self._local_path` in `TransformInput`

Reply:
We also use `self._local_path` in `LightweightTransformInput` which is a subclass. What would you say if we turned `_local_path` into a private property?